File: ios-bluetooth/KhrysalisBluetooth/com/lightningkite/khrysalis/bluetooth/BleServer.shared.swift
//Package: com.lightningkite.khrysalis.bluetooth
//Converted using Khrysalis2

import Foundation
import RxSwift
import RxRelay
import Khrysalis



public protocol BleServer: Disposable {
    
    var characteristics: Dictionary<UUID, Dictionary<UUID, BleCharacteristicServer>> { get }
    
    var advertising: Bool { get set }
}
 
 

public protocol BleClient {
    
    var info: BleDeviceInfo { get }
    
    var connected: Bool { get }
    
    func respond(request: RequestId, data: Data, status: BleResponseStatus) -> Void
    func respond(_ request: RequestId, _ data: Data, _ status: BleResponseStatus) -> Void
    
    func notify(service: UUID, characteristic: UUID, value: Data) -> Void
    func notify(_ service: UUID, _ characteristic: UUID, _ value: Data) -> Void
    
    func indicate(service: UUID, characteristic: UUID, value: Data) -> Void
    func indicate(_ service: UUID, _ characteristic: UUID, _ value: Data) -> Void
}
 

File: android-bluetooth/src/main/java/com/lightningkite/khrysalis/bluetooth/BleServer.shared.kt
package com.lightningkite.khrysalis.bluetooth

import com.lightningkite.khrysalis.bytes.Data
import com.lightningkite.khrysalis.observables.*
import io.reactivex.disposables.Disposable
import java.util.*

//My implementation

interface BleServer: Disposable {
    val characteristics: Map<UUID, Map<UUID, BleCharacteristicServer>>
    var advertising: Boolean
}

interface BleClient {
    val info: BleDeviceInfo
    val connected: Boolean
    fun respond(request: RequestId, data: Data, status: BleResponseStatus)
    /** Does not require a confirmation from the device. **/
    fun notify(service: UUID, characteristic: UUID, value: Data)
    /** Requires a confirmation from the device. **/
    fun indicate(service: UUID, characteristic: UUID, value: Data)
}
